src/test/reporters/base.ts

```diff
@@ -148,41 +202,56 @@ export class BaseReporter implements Reporter  {
   }
 }
 
-export function formatFailure(config: FullConfig, test: TestCase, index?: number, stdio?: boolean): string {
+export function workspaceRelativePath(filePath: string): string {
+  return path.relative(process.env['GITHUB_WORKSPACE'] ?? '', filePath);
+}
+
+export function formatFailure(config: FullConfig, test: TestCase, index?: number, stdio?: boolean, attachment = true): {
+  message: string,
+  annotations: Annotation[]
+} {
   const lines: string[] = [];
-  lines.push(colors.red(formatTestHeader(config, test, '  ', index)));
+  const title = formatTestTitle(config, test);
+  const filePath = workspaceRelativePath(test.location.file);
```

Comment:
Let's inject this via passing it as an argument into formatFailure. That way base does not know about github.